rai/node/node.cpp

```diff
@@ -3718,7 +3718,16 @@ rai::thread_runner::thread_runner (boost::asio::io_service & service_a, unsigned
 			}
 			catch (...)
 			{
+#ifndef NDEBUG
+				/*
+				 * In a release build, catch and swallow the
+				 * service exception, in debug mode pass it
+				 * on
+				 */
+				throw;
+
 				assert (false && "Unhandled service exception");
```

Comment:
Shouldn't we get rid of this assert?

Reply:
Also, in release mode, should we start a replacement thread?

Reply:
Maybe, but that will have to be a different changeset -- we shouldn't generally have unhandled
exceptions reach this point, so if they are thrown from here in debug mode we can find them

Reply:
fwiw, the exceptions I've seen reaching this point in the past were boost log exceptions (not enough arguments for the format string), but all those should've been fixed. Could be others lurking of course.